madanalysis/configuration/weight_configuration.py

```diff
@@ -0,0 +1,303 @@
+            [getattr(w, group) for w in self if getattr(w, group) is not None]
+        )
+
+        if len(unique) == 0:
+            return {}
+
+        group_dict = {v: [] for v in unique}
+        for w in self:
+            if getattr(w, group) is not None:
+                group_dict[getattr(w, group)].append(w)
+
+        return group_dict
+
+    def pdfset(self, pdfid: int) -> List[Weight]:
+        """retreive weights corresponding to one pdf set"""
+        return WeightCollection([w for w in self if w.pdfset == pdfid])
+
+    def get(self, **kwargs) -> List[Weight]:
+        if len(kwargs) == 0:
+            return []
+        collection = []
+        keys = ["muf", "mur", "dynamic_scale", "pdfset", "merging"]
+
+        for weight in self:
+            add = True
+            for key in (key for key in keys if key in kwargs):
+                if getattr(weight, key) != kwargs.get(key):
+                    add = False
+                    break
+            if add:
+                collection.append(weight)
+
+        return WeightCollection(collection)
+
+    @property
+    def pdfsets(self) -> List[int]:
+        """Retreive a list of pdfsets"""
+        return np.unique([w.pdfset for w in self if w.pdfset is not None]).tolist()
+
+    @property
+    def scales(self) -> Dict[Text, List[float]]:
+        """return scale for muf and mur"""
+        muf = np.unique([w.muf for w in self if w.muf is not None]).tolist()
+        muf.sort()
+        mur = np.unique([w.mur for w in self if w.mur is not None]).tolist()
+        mur.sort()
+        return {"muf": muf, "mur": mur}
+
+    @property
+    def has_scale(self) -> bool:
+        """is there any scale variations"""
+        return len(self.scales["muf"]) > 0
+
+    @property
+    def central_scale(self) -> float:
+        """retreive central scale"""
+        scales = self.scales["muf"]
+        return scales[len(scales) // 2]
+
+    def get_scale_vars(self, point: int = 3, dynamic: int = None) -> Tuple:
+        if dynamic is not None:
+            dynamic = dynamic if self.has_dyn_scale(dynamic) else None
+
+        scales = self.scales["muf"]
+        if len(scales) == 5 and point == 3:
+            scales = scales[1:-1]
+        elif len(scales) == 7:
+            if point == 3:
+                scales = scales[2:-2]
+            if point == 5:
+                scales = scales[1:-1]
+        elif len(scales) == 9:
+            if point == 3:
+                scales = scales[3:-3]
+            elif point == 5:
+                scales = scales[2:-2]
+            elif point == 7:
+                scales = scales[1:-1]
+
+        min_scale = min(scales)
+        max_scale = max(scales)
+
+        return (
+            self.get_scale(dynamic=dynamic, muf=min_scale, mur=min_scale),
+            self.get_scale(dynamic=dynamic, muf=max_scale, mur=max_scale),
+        )
+
+    def get_scale(
+        self, dynamic: int = None, muf: float = 1.0, mur: float = 1.0
```

Comment:
I assume the choice on how to calculate scale variation is left to the user. I still need to find how at this stage. I leave this comment for later use, as a bookmark.

Reply:
this class is just to select the location map computation is done when the histogram written in the selection.py file